### PR TITLE
SolidusAdmin conditional routing

### PR DESCRIPTION
**Pull request by elia**
## Summary

https://user-images.githubusercontent.com/1051/236250201-0a3e545f-1191-4f5a-acad-847724cadb3f.mov

It's currently tied to a cookie, but could be based on a Solidus preference or database check etc.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: admin/lib/generators/solidus_admin/install/install_generator.rb [hunk 1]
```diff
@@ -4,7 +4,11 @@ module SolidusAdmin
   module Generators
     class InstallGenerator < Rails::Generators::Base
       def install_solidus_core_support
-        route "mount SolidusAdmin::Engine, at: '/solidus_admin'"
+        route <<~RUBY
+          mount SolidusAdmin::Engine, at: '/admin', constraints: ->(req) {
```
Comment (waiting-for-dev) [inline admin/lib/generators/solidus_admin/install/install_generator.rb:8]: That's exactly the idea I had in mind. Happy you tried it out :slightly_smiling_face:

**Review by waiting-for-dev — APPROVED** [review]

**Comment by waiting-for-dev** [comment]
Why are we generating the routes from the template instead of calling `Spree::Core::Engine.routes.draw do` as with the other Solidus engines? :thinking:

File: admin/lib/generators/solidus_admin/install/install_generator.rb [hunk 2]
```diff
@@ -4,7 +4,11 @@ module SolidusAdmin
   module Generators
     class InstallGenerator < Rails::Generators::Base
       def install_solidus_core_support
```
Comment (waiting-for-dev) [inline admin/lib/generators/solidus_admin/install/install_generator.rb:6]: As we commented offline with @elia, we need to make sure routes are mounted before the ones for `Spree::Core::Engine`.

**Comment by waiting-for-dev** [comment]
> Why are we generating the routes from the template instead of calling `Spree::Core::Engine.routes.draw do` as with the other Solidus engines? thinking

As we also commented offline with @elia, that's a better approach and the one currently used in extensions. We don't monkey patch `solidus_core` routes so we don't get in its way.

File: admin/lib/generators/solidus_admin/install/install_generator.rb [hunk 3]
```diff
@@ -4,7 +4,11 @@ module SolidusAdmin
...
-        route "mount SolidusAdmin::Engine, at: '/solidus_admin'"
+        route <<~RUBY
+          mount SolidusAdmin::Engine, at: '/admin', constraints: ->(req) {
+            req.cookies['solidus_admin'] == 'true'
```
Comment (rainerdema) [inline admin/lib/generators/solidus_admin/install/install_generator.rb:9]: This approach looks promising! 👏

**Review by rainerdema — APPROVED** [review]
💯

**Pull request merged**